unet/unet_model.py

```diff
@@ -64,3 +91,208 @@ def forward(self, x):
         out = self.out_conv(x)
 
         return out
+
+    def setup(self, stage):
+        dataset = OrthogonalPhotoDataset(**self.hparams)
+        val_amount = int(len(dataset) * self.hparams.val_percent / 100)
+        train_amount = len(dataset) - val_amount
+
+        # Development-mode
+        if not os.getenv("PROD"):
+            # Fix the generator for reproducible results
+            self.train_set, self.val_set = random_split(
+                dataset,
+                [train_amount, val_amount],
+                generator=torch.Generator().manual_seed(42),
+            )
+        # Production-mode
+        else:
+            # Random generator
+            self.train_set, self.val_set = random_split(
+                dataset, [train_amount, val_amount]
+            )
+
+        # Disable augmentation for validation set.
+        self.val_set.dataset.set_augmentation(False)
+
+    def train_dataloader(self):
+        return DataLoader(
+            self.train_set,
+            batch_size=self.hparams.batch_size,
+            shuffle=True,
+            num_workers=int(os.getenv("WORKERS", 0)),
+            pin_memory=True,
+        )
+
+    def val_dataloader(self):
+        return DataLoader(
+            self.val_set,
+            batch_size=self.hparams.batch_size,
+            shuffle=False,
+            num_workers=int(os.getenv("WORKERS", 0)),
+            pin_memory=True,
+            drop_last=True,
+        )
+
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(
+            self.parameters(), lr=(self.lr or self.learning_rate)
+        )
+        return [optimizer]
+
+    def loss_funciton(self, input, target):
```

Comment:
Codacy found an issue: [Redefining built-in 'input'](https://app.codacy.com/gh/gil-uav/semantic-image-segmentation/pullRequest?prid=6243237)